docker.go

```diff
@@ -343,6 +356,10 @@ func (c *DockerContainer) Terminate(ctx context.Context) error {
 	c.sessionID = ""
 	c.isRunning = false
 
+	if err := options.Cleanup(); err != nil {
```

Comment:
bug: this `err` definition is actually shadowing the existing variable.
```suggestion
	if err = options.Cleanup(); err != nil {
```